core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala

```diff
@@ -82,4 +83,34 @@ class LogicalPlanGenerator(val explicitDistinct: Boolean = false) extends Genera
     val duplicates = if (setOp.is_all) " ALL" else if (explicitDistinct) " DISTINCT" else ""
     s"(${generate(ctx, setOp.left)}) $op$duplicates (${generate(ctx, setOp.right)})"
   }
+
+  private def generateMerge(ctx: GeneratorContext, mergeIntoTable: MergeIntoTable) = {
+    val target = generate(ctx, mergeIntoTable.targetTable)
+    val source = generate(ctx, mergeIntoTable.sourceTable)
+    val condition = expr.generate(ctx, mergeIntoTable.mergeCondition)
+
+    val matchedActions = mergeIntoTable.matchedActions
+      .map { action =>
+        val conditionText = action.condition.map(cond => s" AND ${expr.generate(ctx, cond)}").getOrElse("")
+        s" WHEN MATCHED$conditionText THEN ${expr.generate(ctx, action)}"
+      }
+      .mkString("")
+
+    val notMatchedActions = mergeIntoTable.notMatchedActions
+      .map { action =>
+        val conditionText = action.condition.map(cond => s" AND ${expr.generate(ctx, cond)}").getOrElse("")
+        s" WHEN NOT MATCHED$conditionText THEN ${expr.generate(ctx, action)}"
+      }
+      .mkString("")
+
+    val notMatchedBySourceActions = mergeIntoTable.notMatchedBySourceActions
+      .map { action =>
+        val conditionText = action.condition.map(cond => s" AND ${expr.generate(ctx, cond)}").getOrElse("")
+        s" WHEN NOT MATCHED BY SOURCE$conditionText THEN ${expr.generate(ctx, action)}"
+      }
+      .mkString("")
+
+    s"MERGE INTO $target USING $source ON $condition$matchedActions$notMatchedActions$notMatchedBySourceActions"
```

Comment:
```suggestion
    s"MERGE INTO $target USING $source ON ${condition}${matchedActions}${notMatchedActions}${notMatchedBySourceActions}"
```
it's a bit more readable this way

Reply:
changed